patroni/config_generator.py

```diff
@@ -0,0 +1,289 @@
+"""patroni --generate-config machinery."""
+import os
+import socket
+import sys
+import yaml
+
+from typing import Any, Dict, Optional
+
+from .config import Config
+from .exceptions import PatroniException
+from .postgresql.config import ConfigHandler
+from .postgresql.misc import postgres_major_version_to_int
+from .utils import get_major_version, patch_config
+
+
+def get_ip() -> str:
+    """
+    Try to get an ip address of the hostname returned by :func:`gethostname`.
```

Comment:
```suggestion
    """Try to get an ip address of the hostname returned by :func:`~socket.gethostname`.
```

Just to keep the pattern of the other docstrings in the module.
About `gethostname`, when generating API documentation IIRC it requires the fully qualified name -- although we can show only the function name in the actual generates docs by using `~`.